### PR TITLE
WIP: Inclusion of netbox.netbox in Ansible 2.10

### LINKED ISSUE #285: netbox_interface no longer exists
##### ISSUE TYPE

 - Bug Report

##### SUMMARY
In https://github.com/netbox-community/ansible_modules/pull/225 `netbox_interface` was deleted. This module previously shipped in Ansible, so it need to continue to exist.
If it's been replaced by another module (though under a different name) then `meta/runtime.yml` needs updating to include ` redirect`

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```

```

### LINKED ISSUE #298: Inclusion of netbox.netbox in Ansible 2.10
This collection will be included in Ansible 2.10 because it contains modules and/or plugins that were included in Ansible 2.9. Please review:

* The list of requirements for inclusion in Ansible: https://github.com/ansible-collections/overview/blob/master/collection_requirements.rst
* The roadmap with all important dates for Ansible 2.10: https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/COLLECTIONS_2_10.rst

## DEADLINE: 2020-08-18

The latest version of the collection available on August 18 will be included in Ansible 2.10.0, except possibly newer versions which differ only in the patch level. (For details, see [the roadmap](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/COLLECTIONS_2_10.rst)). Please release version 1.0.0 of your collection by this date! **If 1.0.0 does not exist, the same 0.x.y version will be used in all of Ansible 2.10 without updates,** and your 1.x.y release will not be included until Ansible 2.11 (unless you request an exception at a [community working group meeting](https://github.com/ansible/community/issues/539) and go through a demanding manual process to vouch for backwards compatibility . . . you want to avoid this!).

## Follow semantic versioning rules

Your collection versioning must follow all [semver rules](https://semver.org/). This means:

* Patch level releases can only contain bugfixes;
* Minor releases can contain new features, new modules and plugins, and bugfixes, but must not break backwards compatibility;
* Major releases can break backwards compatibility.

## Changelogs and Porting Guide

Your collection should provide data for the Ansible 2.10 changelog and porting guide. The changelog and porting guide are automatically generated from ansible-base, and from the changelogs of the included collections. All changes from the `breaking_changes`, `major_changes`, `removed_features` and `deprecated_features` sections will appear in both the changelog and the porting guide. You have two options for providing changelog fragments to include:

1. If possible, use the [antsibull-changelog tool](https://github.com/ansible-community/antsibull-changelog/), which uses the same changelog fragment as the ansible/ansible repository (see the [documentation](https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelogs.rst)).
2. If you cannot use antsibull-changelog, you can provide the changelog in a machine-readable format as `changelogs/changelog.yaml` inside your collection (see the [documentation of changelogs/changelog.yaml format](https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelog.yaml-format.md)).

If you cannot contribute to the integrated Ansible changelog using one of these methods, please provide a link to your collection's changelog by creating an issue in https://github.com/ansible-community/ansible-build-data/. If you do not provide `changelogs/changelog.yml` or a link, users will not be able to find out what changed in your collection from the Ansible changelog and porting guide.

## Make sure your collection passes the sanity tests

Run `ansible-test sanity --docker -v` in the collection with the latest [ansible-base](https://pypi.org/project/ansible-base/) or `stable-2.10` ansible/ansible checkout.

## Keep informed

Be sure you're subscribed to:

* [Changes impacting Collections](https://github.com/ansible-collections/overview/issues/45) to track changes that Collection maintainers should be aware of;
* The Bullhorn, a newsletter for the Ansible developer community, [back issues and how to add content](https://github.com/ansible/community/issues/546).

## Questions and Feedback

If you have questions or want to provide feedback, please see [the Feedback section in the collection requirements](https://github.com/ansible-collections/overview/blob/master/collection_requirements.rst#feedback).

(Internal link to keep track of issues: ansible-collections/overview#102)

### PR DESCRIPTION
**Pull request by FragmentedPacket**
Fixes #285
Fixes #298

This is WIP and meant to try and get everything required to be included in Ansible 2.10 taken care of before 8/18.
Need to incorporate everything [here](https://github.com/ansible-collections/overview/blob/master/collection_requirements.rst)

- [x] Release v1.0.0 of collection
- [x] Tags must be set in `galaxy.yml`
- [x] Dependencies must be set to `>=1.0.0` in `galaxy.yml`
- [x] Set all necessary stuff in `meta/runtime.yml`
- [x] Add `netbox_interface.py` back and add meta/runtime.yml for migration purposes
- [ ] Validate collection passes `ansible-test sanity --docker -v` (may need to exclude some stuff in sanity testing
- [x] Generate proper changelog in accordance to Ansible recommendation

### CODE REVIEW

**Comment by FragmentedPacket** [comment]
@felixfontein Do I need to populate the `changelogs/changelog.yaml` with all my release notes? Or is the current one in this PR fine for 2.10?

Is there a way we can manage some ignores for the errors you see in the latest run of my CI for Ansible 2.10 preparation?

I'm trying to skip all Python2 tests as this stuff doesn't support anything below Python3? And possible to not test whole directories such as my `tests/` directory?

**Comment by FragmentedPacket** [comment]
@gundalow Is the `meta/runtime.yml` look correct to you now?

**Comment by gundalow** [comment]
Yes, thank you.

File: changelogs/.plugin-cache.yaml [hunk 1]
```diff
@@ -0,0 +1,278 @@
+plugins:
```
Comment (gundalow) [inline changelogs/.plugin-cache.yaml:1]: Please add this to `.gitignore`and remove this file from the PR

**Comment by felixfontein** [comment]
@FragmentedPacket changelogs/changelog.yaml should contain all changes made since the state of the modules/plugins in Ansible 2.9.0.

**Comment by FragmentedPacket** [comment]
@felixfontein Thank you. Do the Sanity tests have to pass completely for the launch?

File: changelogs/.plugin-cache.yaml (hunk 1 above)
Comment (FragmentedPacket) [reply changelogs/.plugin-cache.yaml:1]: I will push that up today. Thank you!

**Comment by felixfontein** [comment]
@FragmentedPacket we won't actively test that, but we recommend that they should all pass. (You can use the ignore-2.x.txt files for errors you want to fix later, or which are wrongly reported.)

**Comment by FragmentedPacket** [comment]
@gundalow @felixfontein Does this incorporate everything needed before 8-18? I've already cut 1.0.0 release as well.

I will definitely get the sanity tests fixed sooner than later, but want to make sure to meet the deadline.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+        netbox_url: http://netbox-demo.org:32768
+        netbox_token: 0123456789abcdef0123456789abcdef01234567
+        data:
+          name: Test66
+          device_type:
+            id: "{{ some_jinja_variable }}"
+          device_role: Core Switch
+          site: Test Site
+          status: Staged
+        state: present
```
Comment (felixfontein) [inline CHANGELOG.rst:53]: The code block will look a lot better if you use `::` to format it (https://docutils.sourceforge.io/docs/user/rst/quickstart.html#preformatting-code-samples):
```suggestion

  ::

      tasks:
        - name: Create device within NetBox with only required information
          netbox_device:
            netbox_url: http://netbox-demo.org:32768
            netbox_token: 0123456789abcdef0123456789abcdef01234567
            data:
              name: Test66
              device_type:
                id: "{{ some_jinja_variable }}"
              device_role: Core Switch
              site: Test Site
              status: Staged
            state: present
```
You'll need to update the entry in `changelog.yaml` correspondingly and run `antsibull-changelog generate` to update the .rst file.

File: tests/sanity/ignore-2.10.txt [hunk 3]
```diff
@@ -1,2 +1,4 @@
 plugins/modules/netbox_interface.py import-3.8!skip # Module is deprecated, so will not be updated for 2.10 compatibility
 plugins/modules/netbox_interface.py validate-modules!skip
+plugins/modules/* compile-2.6!skip # Collection supports Python 3.6+
```
Comment (felixfontein) [inline tests/sanity/ignore-2.10.txt:3]: Wildcards don't work unfortunately, you have to list every file individually.

File: CHANGELOG.rst [hunk 4]
```diff
@@ -35,9 +35,16 @@ Minor Changes
...
+  requires querying an endpoint is now done by making it a dictionary with
+  an ``id`` key. The previous behavior was to just pass in an integer and
+  it was converted when normalizing the data, but some people may have names
+  that are all integers and those were being converted erroneously so we made
+  the decision to change the method to convert to an integer for the NetBox
+  API.
+
+  ::
+
   tasks:
```
Comment (felixfontein) [inline CHANGELOG.rst:48]: You need to indent this, otherwise the `::` doesn't do anything.

**Pull request merged**